src/game/level_update.c

```diff
@@ -743,17 +743,21 @@ s16 level_trigger_warp(struct MarioState *m, s32 warpOp) {
                 break;
 
             case WARP_OP_WARP_FLOOR:
+                if (m->floor->type == SURFACE_WARP && (m->floor->force != 0x00)) {
+                    sSourceWarpNodeId = m->floor->force;
+                } else {
                 sSourceWarpNodeId = WARP_NODE_WARP_FLOOR;
```

Comment:
needs indentation

Reply:
indent these fucking nuts in your mouth (i agree)